version: '2.2'
services:

  db:
    restart: always
    image: postgres:latest
    networks:
      - db_network
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "postgres"]
    ports:
      - 5432:5432
    environment:
      POSTGRES_PASSWORD: 'postgres'
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./data_munging_tools/csv_to_SQL.sh:/opt/data_munging_tools/csv_to_SQL.sh
      - ./data_sets/ufo-sightings.csv:/opt/data_sets/ufo-sightings.csv
    # command:
    #   - /opt/data_munging_tools/csv_to_SQL.sh

  flask:
    restart: always
    image: xcom:latest
    build: .
    networks:
      - external_network
      - db_network
    healthcheck:
        test: curl --fail http://localhost:5000/ping || exit 1
        interval: 10s
        timeout: 2s
        retries: 5
    expose:
      - 5000
    ports:
      - 5000:5000
    links:
      - "db:database"
volumes:
  postgres_data:

networks:
  external_network:
    driver: bridge
    ipam:
      driver: default
  db_network:
    internal: true
